### DISCUSSION TITLE
how to remap some the lsp keymaps? how to setup which-key for lsp?

### DISCUSSION THREAD

**Discussion by davidgao7** [LazyVim/LazyVim#2880] [Q&A] [answered]
I found some of the lsp keymaps is overwritten by my copilot setting, how can I reset the lsp keymap and also display hint on which-key?
<img width="1118" alt="Screenshot 2024-03-29 at 12 01 04 PM" src="https://github.com/LazyVim/LazyVim/assets/39502459/5c916de6-cfe7-4c7f-ad86-3323741223ad">

**Comment by dpetka2001** [comment]
The docs have an [explicit section](https://www.lazyvim.org/plugins/lsp#%EF%B8%8F-customizing-lsp-keymaps) about how to change LSP keymaps. And if they have an adequate `desc`, it will be picked up by `which-key` I believe.

**Reply by davidgao7** [reply]
Thank you!
I want to make each my config file structured, and the example feels like a plugin

should I custom it in keymaps.lua in my ~/.config/nvim/lua/config/  or can I write a new lua file in ~/.config/nvim/lua/plugins/?

<img width="1131" alt="Screenshot 2024-03-29 at 12 14 20 PM" src="https://github.com/LazyVim/LazyVim/assets/39502459/6c11a9aa-9f6a-4b5b-8014-810ee5358d9d">

**Reply by dpetka2001** [reply, marked as answer]
In a file in `~/.config/nvim/lua/plugins/` directory.
